### ISSUE TITLE
NullAway 0.13.6 no longer flags uninitialized JPA property missing @Nullable

### ISSUE THREAD

**Issue by jeffrey-easyesi** [uber/NullAway#1602]
```java
import jakarta.persistence.*;
@Entity
@org.jspecify.annotations.NullMarked
public class TestEntity {
    private String name;
    private String extraInfo;

    public TestEntity() {} // for JPA
    public TestEntity(String name) { this.name = name; }

    @Id
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getExtraInfo() { return extraInfo; }
    public void setExtraInfo(String s) { this.extraInfo = s; }
}
```

NullAway 0.13.4 correctly warns on the uninitialized non-null field `extraInfo`, which ought to be `@Nullable`.
(It also complains about the no-arg constructor, but that's easily fixed with `NullAway:ExternalInitAnnotations=jakarta.persistence.Entity`.)

NullAway 0.13.6 allows `extraInfo` to be unannotated, even though it probably isn't set for all objects.

**Comment by msridhar** [comment]
Hi @jeffrey-easyesi this change was deliberate. I assumed the `@Entity` annotation would mean that the field would get initialized by JPA. Is your concern here that there is no warning for the other constructor that takes an argument?

**Comment by jeffrey-easyesi** [comment]
Yes - a typical entity class for us has both
* no-arg constructor for the JPA framework to call when deserializing existing entities from the database
* other constructor(s) called from application code to create a new entity, i.e. `entityManager.persist(new TestEntity(name));`

**Comment by msridhar** [comment]
Thanks, @jeffrey-easyesi, let me think about how we can still report warnings for constructors with arguments in these cases.
